src/smt/smt_engine.cpp

```diff
@@ -4795,12 +4795,16 @@ void SmtEngine::checkSynthSolution()
   Notice() << "SmtEngine::checkSynthSolution(): checking synthesis solution" << endl;
   map<Node, Node> sol_map;
   /* Get solutions and build auxiliary vectors for substituting */
-  d_theoryEngine->getSynthSolutions(sol_map);
-  if (sol_map.empty())
+  if (!d_theoryEngine->getSynthSolutions(sol_map))
   {
     Trace("check-synth-sol") << "No solution to check!\n";
     return;
   }
+  if (sol_map.empty())
+  {
+    Trace("check-synth-sol") << "Got empty solution!\n";
+    return;
+  }
```

Comment:
Shouldn't this be an assertion that if we got here. i.e. if we successfully got solutions for all conjectures, then the map is not empty? Or are we allowing conjectures that have no functions-to-synthesize?

Reply:
This preserves the previous behavior, although I will think a bit whether it should change.